src/qt/optionsdialog.cpp

```diff
@@ -50,6 +51,10 @@ OptionsDialog::OptionsDialog(QWidget *parent, bool enableWallet) :
 #ifndef USE_UPNP
     ui->mapPortUpnp->setEnabled(false);
 #endif
+    connect(this, &QDialog::accepted, [this](){
+        QSettings settings;
+        model->node().mapPort(settings.value("fUseUPnP").toBool());
```

Comment:
In commit "gui: Apply port mapping changes on dialog exit" (58e8364dcdc4e57b0caac09f8402e6535301de9b)

@hebasto It seems like this commit has a bug and will incorrectly overwrite command line and config file `-upnp` and `-natpmp` options. I think this might have happened before, but previously only if these settings were changed, now if the dialog is just opened and closed.

Would you have an opinion on reverting this change? Aside from the bug, it seems simpler to go back to fully applying settings in OptionsModel, instead of using half-using OptionsModel to apply the setting, and then reading the setting later in a different code path and QSettings instance in dialog code outside of OptionsModel.

(Encountered this while rebasing #15936)

Reply:
@ryanofsky

You are right.

> It seems like this commit has a bug and will incorrectly overwrite command line and config file `-upnp` and `-natpmp` options.

The concerns about user-friendly and correct behavior were raised in #18184. Now users are able to switch `-upnp` and `-natpmp` on/off on-the-fly via the GUI. Should we drop this feature?

> I think this might have happened before, but previously only if these settings were changed, now if the dialog is just opened and closed.

Exiting the dialog via the "Cancel" push button or pressing the "Esc" key should not trigger that code path.

Anyway, I'll happy to revert (or enhance) that change in a proper way.